clang/lib/Driver/Driver.cpp

```diff
@@ -769,6 +769,15 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
     }
     return SYCLArg;
   };
+
+  // Emit an error if c-compilation is forced in -fsycl mode
+  if (HasValidSYCLRuntime && C.getInputArgs().hasArg(options::OPT_x))
+    for (StringRef XValue : C.getInputArgs().getAllArgValues(options::OPT_x)) {
```

Comment:
This seems sufficient:
```suggestion
  if (HasValidSYCLRuntime)
    for (StringRef XValue : C.getInputArgs().getAllArgValues(options::OPT_x)) {
```

Reply:
Yes I thought so too but in majority cases we don't have -x options so thought this offers better performance. I can make suggestion as you pointed if the performance drop expectation is negligible.

Reply:
Actually, I don't see why an extra iteration over the arguments' list would bring better performance. We'll still have to traverse the full collection once, and the extra check itself requires such a traversal just like all values extraction.
Another thing that's occurred to me right now: would it make sense to get only the last argument value, which is supposed to override the preceding ones?
```suggestion
  if (HasValidSYCLRuntime)
    StringRef XValue = C.getInputArgs().getLastArg(options::OPT_x);
    if (!XValue.empty()) {
```

Reply:
Fine with me if @mdtoguchi agrees

Reply:
Thank you AGindinson for the approval. I will ask Mike Toguchi for the review